packages/ingest/src/lock.ts

```diff
@@ -31,30 +43,54 @@ async function checkOldLocks(bucket, locks = []) {
 * @param {string} providerName - The provider name
 * @returns {integer} - Number of current locks in the bucket
 **/
-async function countLock(bucket, providerName) {
+export async function countLock(
+  bucket: string,
+  providerName: string
+): Promise<number> {
   const s3Objects = await listS3ObjectsV2({
     Bucket: bucket,
     Prefix: `${lockPrefix}/${providerName}`
   });
-  return checkOldLocks(bucket, s3Objects);
+
+  if (s3Objects === undefined) return 0;
+
+  const locks = <Lock[]>s3Objects.filter(({ Key }) => Key !== undefined);
+
+  return checkOldLocks(bucket, locks);
 }
 
-function addLock(bucket, providerName, filename) {
-  return s3PutObject({
+async function addLock(
+  bucket: string,
+  providerName: string,
+  filename: string
+): Promise<void> {
+  await s3PutObject({
     Bucket: bucket,
     Key: `${lockPrefix}/${providerName}/${filename}`,
     Body: ''
   });
 }
 
-function removeLock(bucket, providerName, filename) {
-  return deleteS3Object(
+export async function removeLock(
+  bucket: string,
+  providerName: string,
+  filename: string
+): Promise<void> {
+  await deleteS3Object(
```

Comment:
Oof - I don't love relying on the global state of`lockPrefix` in these methods. Do you mind at least making that a function returning the same value that is used in these functions?

Reply:
`lockPrefix` isn’t global, it’s a constant declared in this module.

Reply:
Yeah I meant global to the scope of the module. I still think it'd be preferable to change for the reasons globals are generally discouraged, but I won't hold up the PR on it